app/helpers/course/assessment/submission/submissions_helper.rb

```diff
@@ -14,14 +14,14 @@ def comments_container_id(submission_question)
     "course_assessment_submission_question_#{submission_question.id}_comments"
   end
 
-  # Return the last attempted answer based on the status of current submission.
-  # previous attempt if submission is in attempting state.
-  # current attempt if submission is in submitted, graded or published state.
+  # Return the graded last attempted answer based on the status of current submission,
+  # or the last attempt if there are no evaluated or graded answers.
   #
   # @return [Course::Assessment::Answer]
   def last_attempt(answer)
     submission = answer.submission
     attempts = submission.answers.from_question(answer.question_id)
-    submission.attempting? ? attempts[-2] : attempts[-1]
+    graded_or_evaluated = attempts.select { |x| ['evaluated', 'graded'].include?(x.workflow_state) }
```

Comment:
Use `x.evaluated? || x.graded?` instead of strings.  Strings could give unexpected results when we change the workflow state.